WordPress/src/main/java/org/wordpress/android/viewmodel/history/HistoryViewModel.kt

```diff
@@ -126,10 +145,15 @@ class HistoryViewModel @Inject constructor(
     }
 
     private fun fetchRevisions() {
-        _listStatus.value = HistoryListStatus.FETCHING
-        val payload = FetchRevisionsPayload(post, site)
-        uiScope.launch {
-            dispatcher.dispatch(PostActionBuilder.newFetchRevisionsAction(payload))
+        if (networkUtils.isNetworkAvailable()) {
+            _listStatus.value = HistoryListStatus.FETCHING
+            val payload = FetchRevisionsPayload(post, site)
+            uiScope.launch {
+                dispatcher.dispatch(PostActionBuilder.newFetchRevisionsAction(payload))
+            }
+        } else {
+            _listStatus.value = HistoryListStatus.NO_NETWORK
+            createRevisionsList(emptyList())
```

Comment:
This part was odd to me. It looks like if the `onPullToRefresh()` method is called, we would be resetting the list to empty if there is no internet connection. However, we never get to this point because `HistoryListFragment. checkConnectionAndPullToRefresh` checks for the internet connection before calling `onPullToRefresh()`. 

I think there shouldn't be a need for the `Fragment` to check for the internet connection. The `ViewModel` should decide what happens in regards to that. With that said, I'm not quite sure how big of a refactoring that would entail so I'm fine with this for now. 😁

Reply:
>I think there shouldn't be a need for the Fragment to check for the internet connection. The ViewModel should decide what happens in regards to that. 

💯 I'd consider moving the `checkConnectionAndPullToRefresh` logic into the VM.

Reply:
Updated in 5eea3b6

Reply:
It's looking good. 👌 Thanks for changing that!